libs/common/tests/context_builder_test.cpp

```diff
@@ -176,4 +175,70 @@ TEST(ContextBuilderTests, AccessKindBuilderMultipleTimes) {
     EXPECT_TRUE(context.Get("user", "isCat").AsBool());
 }
 
+TEST(ContextBuilderTests, AddAttributeToExistingContext) {
+    auto context = ContextBuilder()
+                       .Kind("user", "potato")
+                       .Name("Bob")
+                       .Set("city", "Reno")
+                       .SetPrivate("private", "a")
+                       .Build();
+
+    auto builder = ContextBuilder(context);
+    if (auto updater = builder.Kind("user")) {
```

Comment:
@mmrj Adding "state" and "sneaky" to the context.

Reply:
nice!

Reply:
could I also do:
```
updater->Set("city", "Winnemucca")
```
and overwrite the `Reno` value?

Reply:
Yes.